synspec.py

```diff
@@ -0,0 +1,50 @@
+from synphot.observation import Observation
+import astropy.units as u
+import numpy as np
+
+
+def from_R(source_spectrum, bandpass, R, waverange, force='none'):
+    """
+    Returns a spectrum with bins defined by a spectral resolution
+    R and a given range of wavelengths. The bin width varies over the
+    range of wavelengths such that the ratio of the wavelength to the
+    bin width at that wavelength is equal to the given R.
+    """
+    first_wl = waverange[0].to(u.angstrom).value
+    binset = np.array([first_wl])
+    binwidth = first_wl / R
+
+    while binset[-1] <= waverange[-1].to(u.angstrom).value:
```

Comment:
I haven't though very hard about this yet, but is there a way to write this without a `while` loop, to keep it speedy?

Reply:
Will think more about this!